pandas/core/groupby/generic.py

```diff
@@ -1185,15 +1185,33 @@ def nsmallest(
     def idxmin(
         self, axis: Axis | lib.NoDefault = lib.no_default, skipna: bool = True
     ) -> Series:
-        result = self._op_via_apply("idxmin", axis=axis, skipna=skipna)
-        return result.astype(self.obj.index.dtype) if result.empty else result
+        if axis is not lib.no_default:
+            axis = self.obj._get_axis_number(axis)
+            self._deprecate_axis(axis, "idxmax")
+        if axis == 1:
+            result = self._op_via_apply("idxmax", axis=axis, skipna=skipna)
+            if result.empty:
+                result = result.astype(self.obj.index.dtype)
+        else:
+            alt = lambda x: x.idxmin(skipna=skipna)
```

Comment:
what cases need the python-path fallback?

Reply:
object

Reply:
avoidable if you use numeric_object_t in libgroupby?

Reply:
Nice! That worked!